libmscore/scoretree.cpp

```diff
@@ -0,0 +1,831 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "accidental.h"
+#include "ambitus.h"
+#include "arpeggio.h"
+#include "articulation.h"
+#include "barline.h"
+#include "beam.h"
+#include "bracket.h"
+#include "bsymbol.h"
+#include "chord.h"
+#include "duration.h"
+#include "fret.h"
+#include "glissando.h"
+#include "hook.h"
+#include "iname.h"
+#include "ledgerline.h"
+#include "lyrics.h"
+#include "measure.h"
+#include "measurenumber.h"
+#include "note.h"
+#include "page.h"
+#include "rest.h"
+#include "score.h"
+#include "segment.h"
+#include "spacer.h"
+#include "spanner.h"
+#include "staff.h"
+#include "stafflines.h"
+#include "stem.h"
+#include "stemslash.h"
+#include "system.h"
+#include "systemdivider.h"
+#include "textframe.h"
+#include "tie.h"
+#include "tremolo.h"
+#include "trill.h"
+#include "tuplet.h"
+
+namespace Ms {
+//---------------------------------------------------------
+//   Score
+//---------------------------------------------------------
+
+ScoreElement* Score::treeParent() const
+{
+    return nullptr;  // Score is root node
```

Comment:
Maybe for `Score` parent is the `MasterScore`, and the `MasterScore` should be the root?

Reply:
There is still some confusion about what Score and MasterScore actually mean, so my vote is to leave it how it is for now and then wait to see if it causes problems.

Reply:
It seems like for now `Score` and `MasterScore` actually point to the same object.